organized requirements into sections where possible; else, just gave everything its own citable section and R- requirement (#6)
danwing/requirements-https-local-domains: draft-nygren-requirements-https-local-domains.md

appear identical to an attack.  This would be unacceptable.

> R-UNIQUE-NAME: The system MUST have a way to uniquely identify
  servers.

## Cryptographic Binding

A server's name has to be mapped to its cryptographic identity.

> R-BINDING: The Web Origin MUST be cryptographically bound to one or
  more key pairs, where the private keying material is on the service
  endpoint and where an attacker without the private key(s) is unable
  to access any state associated with the Web Origin.

A client has to be able to validate the name maps to the cryptographic
identity.

> R-VALIDATE: Clients MUST be able to cryptographically validate that
  the authenticating server matches the identity in the URI / Web
  Origin.

Web browsers and modern users both expect a URI.

> R-URI: It MUST be possible to construct a URI that encapsulates a
  Web Origin and its cryptographically-bound identity information.

## Abstract Naming

Using IP addresses in names is problematic if the server's IP address
changes due to ISP renumbering or internal network DHCP server
reconfiguration.

> R-ABSTRACT: The solution SHOULD abstract names from IP addresses.

> Any given name should be able to have a mixture of IPv4, IPv6

Reviewer: ```suggestion
Any given name should be resolvable to a mixture of IPv4, IPv6
```

Link-Local (on an Interface), IPv6 ULA, and IPv6 Globally-Routable
addresses. Operating a local DNS is beyond the scope of many
administrators, so being able to advertise the server using
{{?DNS-SD=RFC6763}} is necessary.

> R-DNS-SD: The name MUST be advertisable using {{?DNS-SD=RFC6763}}

## Avoid Central Authority

A solution needs to be self-contained and not use the central
authority of PKIX.

> R-AVOID-CENTRAL: A solution SHOULD NOT (MUST NOT?) rely on central
  trust hierarchy.

Vendors go out of business or lose interest in continuing to service
old products. The products may still be operational.

> R-AVOID-VENDOR: A solution SHOULD NOT (MUST NOT?) have continued
  reliance on a service operated by a vendor, including if the device
  is reset to factory defaults (e.g., reset for troubleshotting or
  because sold).

## Multiple Application Protocols

> R-MULT-APP: A solution MUST support various application-level

>   protocols such as HTTPS, IPPS, DoH, DoT, SMB over QUIC, and so on.

Reviewer: ```suggestion
  protocols, especially HTTPS, IPPS, DoH, DoT, ande SMB over QUIC.
```

## Cryptographic Agility

> R-AGILITY: A solution SHOULD support crypto agility (such as
  supporting more than one active key type).

## TLS Server Name Indication

> R-TLS-SNI: A solution SHOULD support TLS SNI so a server knows which
key pair/cert is expected.

## Localhost

> R-LOCALHOST: A solution SHOULD support localhost uses as well (e.g.,
for sending a user to connect to a local service)

## W3C Private Network Access

> R-PNA: A solution SHOULD integrate well with an evolution of
{{w3c-pna}} and both allow for an improved model there but should also
provide more robust solutions to vulnerabilities that it tries to
address

## Constrain to Local Resources

> R-LOCAL: A solution SHOULD be constrained to .local and .internal.

> Discuss: MAY constrain to the DHCP domain-search value??  Should we
also allow any arbitrary name if the IP address is local (RFC1918
address), too?

> ## Technical Requirements

Reviewer: ```suggestion
## Misc.
```
